app/domain/authentication/authn_jwt/id_from_decoded_token_provider.rb

```diff
@@ -0,0 +1,58 @@
+module Authentication
+  module AuthnJwt
+    # Class for providing jwt identity from the decoded token from the field specified in a secret
+    class IdentityFromDecodedTokenProvider < IdentityProviderInterface
+
+      attr_accessor :secret_fetcher
+
+      def initialize(authentication_parameters)
+        @resource_id = authentication_parameters.authenticator_resource_id
+        @decoded_token = authentication_parameters.decoded_token
+        @secret_fetcher = Conjur::FetchRequiredSecrets
+      end
+
+      def provide_jwt_identity
+        token_field_name = fetch_token_field_name
+        if @decoded_token.include?(token_field_name)
+          @decoded_token[token_field_name]
+        else
+          raise Errors::Authentication::AuthnJwt::NoSuchFieldInToken.new(
+            token_field_name
+          )
+        end
+      end
+
+      # Checks if variable that defined from which field in decoded token to get the id is configured
+      def identity_available?
+        resource = ::Resource[token_id_field_resource_id]
+        if resource
+          return true
+        end
+        false
```

Comment:
That if is weird...
I interpret it like this:
```
if (something is true)
  return true
else
  return false
end
```
wouldn't it be simpler to just `return resource.present?` or something similar?
Also, I think that the class `::Resource` has something similar to this.
look for same implementations in other places in the code.

Reply:
Done